### PR TITLE
Add option the popup to close on mouseleave

### LINKED ISSUE #123: The popup to close when you move the mouse away from its area
Currently, once you mouseover the notification counter and the popup appears,
the only way to close it is to click outside the popup.

My suggestion is the popup to close when you move the mouse away from its area.
In other words, the popup to remain visible only while the cursor is near its area.

Thank you

(extension 21.3.9 in Chrome 89/Firefox 87)

### PR DESCRIPTION
**Pull request by darkred**
Closes #123 

## Test URLs

https://github.com/

## Screenshot

---

## Note to reviewers

I use `.removeAttribute('open')` to close the dropdown.

### CODE REVIEW

**Comment by fregante** [comment]
Thank you! I’ll make sure it works correctly soon

**Comment by darkred** [comment]
I made a change so that the popup close to be canceled if you re-mousenter quick enough, within 1 sec (1000 ms).
(and updated the screenshot in the OP).

File: source/options.html [hunk 1]
```diff
@@ -15,6 +15,12 @@
 			Show only participating notifications
 		</label>
 	</p>
+	<p>
+		<label>
+			<input type="checkbox" name="closeOnMouseleave">
+			The popup to close when you move the mouse away from its area
```
Comment (jdreesen) [inline source/options.html:21]: This sounds strange to me. Maybe write it like this?

```suggestion
			Automatically close the popup on mouse leave
```

Comment (darkred) [reply source/options.html:21]: I agree my wording was not a very good choice.

But, doesn't "on mouse leave" sound like a technical/programming term? IMHO it's not self-explanatory enough. 
Maybe change it to the following instead?

> Automatically close the popup when mouse is moved away

What do you think?

**Comment by darkred** [comment]
Hi @fregante 
Is there any way I can assist you about completing this PR ? 

&nbsp;
Some notes:
- About the popup closing to be canceled if you re-mousenter quick enough,
the delay should be smaller,  500mec - having it as 1000msec (= 1sec) seems too much for me.
- I've noticed that if I click "Mark as done" and then move the mouse away so that the popup gets closed,
then if I move the mouse to reopen the popup, clicking "Mark as done" 
unfortunately causes the relevant "Marked x notifications as read" message to be displayed multiple times: e.g.

So, I tried changing `removeAttribute()` into `remove()` the popup altogether + reopening it (see code below) 
but unfortunately the issue remains:

    <details>
    <summary>using <code>remove()</code></summary>
    
    ```js
    
    		if (options.closeOnMouseleave) {
    			let timer;
    			select('.NPG-dropdown').addEventListener('mouseleave', () => {
    				// timer = setTimeout(closeDropdown, 500);
    				timer = setTimeout(() => {closeDropdown(); createNotificationsDropdown();}, 500);
    			});
    			select('.NPG-dropdown').addEventListener('mouseenter', () => {
    				clearTimeout(timer);
    			});
    		}
    
    // ...
    
    function closeDropdown() {
    	// select('.NPG-container[open]').removeAttribute('open');
    	select('.NPG-container').remove();
    }
    ```
    
    </details>

File: source/options.html (hunk 1 above)
Comment (darkred) [reply source/options.html]: I used my new wording.

**Comment by darkred** [comment]
> Thank you! I’ll make sure it works correctly soon

@fregante Have you changed your mind? Do you find this PR is not salvageable? Should I close this?

**Comment by fregante** [comment]
I've been on Safari for a year and I haven't used some of my extensions since 😅 so you can just leave this open until someone reviews it

**Comment by tjx666** [comment]
Welcome to contribute here: https://github.com/tjx666/github-notifications-preview/tree/main

**Comment by darkred** [comment]
@fregante Thanks for the merge. Just wanted to note, though, that the problem with my code that I reported [above](https://github.com/tanmayrajani/notifications-preview-github/pull/125#issuecomment-854889625) still exists (the additional multiple `Marked 0 notifications as read`). 
<details>
<summary>Screenshot:</summary>


</details>

&nbsp;
Is it ok if I open an issue for reference? (I haven't managed to fix this problem myself). Or, it's better to revert the merge altogether, so someone else can provide a fully working PR?

**Comment by fregante** [comment]
If this PR introduced the bug, just reuse the closing logic that we already had: `select('.modal-backdrop').click();`

You can place this line in `closeDropdown` and replace the old `select('.modal-backdrop').click();` with `closeDropdown()` as well.

Note that we're still waiting for this to be released though

**Comment by darkred** [comment]
Thanks a lot for the guidance. So, I tried your suggestions but I noticed that the `.modal-backdrop` selector no longer applied (you had used it 7 years ago [here](https://github.com/tanmayrajani/notifications-preview-github/issues/50)). If you could provide me with a working one, I'd gladly make a new PR. (I apologize for asking for your help once again, but I can't seem to find a working one myself).

> Note that we're still waiting for this to be released though

For Firefox it was released alright.

**Comment by fregante** [comment]
Check how it's already working. Clicking somewhere manually closes the dropdown, so you can see what the new selector is.

**Comment by darkred** [comment]
The selector I've found is `details.NPG-container[open] > summary` and it works, but unfortunately the duplicate additional multiple `Marked 0 notifications as read`) still occur.
Here are the 2 changes in total:


<details>
<summary>and here is a screen capture of the issue:</summary>


</details>

**Comment by fregante** [comment]
It's not an issue with this PR. It can be fixed separately:

- https://github.com/tanmayrajani/notifications-preview-github/issues/107

PR welcome with the change I suggested anyway

**Comment by fregante** [comment]
I just submitted this version to the chrome web store. They should publish it within 24 hours

**Pull request merged**